nx/source/runtime/devices/socket.c

```diff
@@ -1580,6 +1581,29 @@ int getnameinfo(const struct sockaddr *sa, socklen_t salen,
     return gaie;
 }
 
+
+long gethostid(void) {
+	Result rc;
+	IGeneralService igs;
+	rc = CreateGeneralService(&igs);
```

Comment:
You leak a handle here, by creating an instance of IGeneralService but never closing it.